src/main/java/de/tum/in/www1/artemis/repository/metis/AnswerPostRepository.java

```diff
@@ -22,20 +25,34 @@ public interface AnswerPostRepository extends JpaRepository<AnswerPost, Long> {
 
     List<AnswerPost> findAnswerPostsByAuthorId(long authorId);
 
+    @Query("""
+                SELECT answerPost
+                FROM AnswerPost answerPost
+                    LEFT JOIN FETCH answerPost.post post
+                    LEFT JOIN FETCH post.conversation
+                WHERE answerPost.id = :answerPostId
+            """)
+    Optional<AnswerPost> findWithPostById(@Param("answerPostId") Long answerPostId);
+
+    default AnswerPost saveAndReload(AnswerPost answerPost) {
+        Long id = save(answerPost).getId();
+        return findWithPostById(id).orElseThrow();
```

Comment:
Do we really need to reload the whole entity or is it enough if we override certain relations with the previous state?
This obviously won't work with cascading relations.